### PR TITLE
Update indexing.py

### PR DESCRIPTION
**Pull request by rahul24629**
Fix silent failure for boolean sequence indexing

### CODE REVIEW

**Comment by jakevdp** [comment]
Can you say more about the goal of this change, and give an example of code that behaves incorrectly without this change?

**Comment by rahul24629** [comment]
Thanks for taking a look!

The goal of this change is to prevent a silent failure mode when boolean indices
are provided as Python sequences (e.g. tuples or lists), which are currently
classified as BOOLEAN indices but are not concrete during tracing.

For example, without this change the following code behaves incorrectly without
raising an error:

```python
import jax
import jax.numpy as jnp
from functools import partial

@partial(jax.jit, static_argnums=(1,))
def f(x, mask):
    return x.at[mask].set(0)

x = jnp.array([1., 2., 3., 4.])
mask = (True, False, True, False)

f(x, mask)  # silently returns the original array

**Comment by jakevdp** [comment]
I see – did you see my reply on this issue? https://github.com/jax-ml/jax/issues/34127#issuecomment-3702305917

Given NumPy's indexing semantics (which JAX follows), returning the original array is the correct behavior here, so this does not indicate any need for a fix.

**Comment by rahul24629** [comment]
Yes, I see it now — thanks for the clarification.

I understand that this behavior follows NumPy’s indexing semantics and is
working as intended. I’ll close the PR since no fix is needed.

Thanks for taking the time to explain!
